keras_nlp/tokenizers/word_piece_tokenizer.py

```diff
@@ -199,13 +200,35 @@ def __init__(
 
         super().__init__(**kwargs)
 
+        if vocabulary_size != None and isinstance(vocabulary_size, int) == False:
+            raise ValueError(
+                f"Vocabulary_size expected type int, given {type(vocabulary_size)}"
+            )
+        
         if isinstance(vocabulary, str):
-            self.vocabulary = [
-                line.rstrip() for line in tf.io.gfile.GFile(vocabulary)
-            ]
+            if vocabulary_size is None:
```

Comment:
We should move all of this logic out of the file case. Setting a vocabulary_size should also truncate a vocabulary that was passed as a list. There should be no difference in the functionality (or the truncation error message) if vocab is a file or vocab is a list.

Totally fine to just do this as a post process, where we read the entire file, and do the truncation afterward.